packages/vue/tsup.config.ts

```diff
@@ -0,0 +1,31 @@
+import autoPropsPlugin from '@vue.ts/tsx-auto-props/esbuild';
+import { defineConfig, type Options } from 'tsup';
+
+import { name, version } from './package.json';
+
+type EsbuildPlugin = NonNullable<Options['esbuildPlugins']>[number];
+
+export default defineConfig(() => {
+  return {
+    clean: true,
+    entry: ['./src/index.ts'],
+    format: ['esm'],
+    bundle: true,
+    sourcemap: true,
+    minify: false,
+    dts: true,
+    esbuildPlugins: [
+      // Automatically generates runtime props from TypeScript types/interfaces for all
+      // control and UI components, adding them to Vue components during build via
+      // Object.defineProperty
+      autoPropsPlugin({
+        include: ['**/*.ts'],
+      }) as EsbuildPlugin,
```

Comment:
Instead of redeclaring all props for all of available components, this plugin will generate a runtime prop using the type prop.

So this component:

<img width="710" alt="Screenshot 2024-11-04 at 2 15 55 PM" src="https://github.com/user-attachments/assets/6a23b11d-8072-4706-8c8d-74196573c3b3">

will turn into this when built:

<img width="467" alt="Screenshot 2024-11-04 at 9 59 17 AM" src="https://github.com/user-attachments/assets/58d9162d-e8fa-41c7-b676-48542a162c67">

So without this plugin, we will have to manually add runtime props to each component, and make sure they will always match.